python/src/aac/plugins/first_party/active_context/active_context_impl.py

```diff
@@ -0,0 +1,161 @@
+            raise PluginError(f"Could not add {file} because it doesn't exist")
+
+        active_context.add_definitions_from_uri(file)
+        return f"Successfully added {file} to the active context"
+
+    with plugin_result(plugin_name, add_file_if_exists) as result:
+        return result
+
+
+def reset_context() -> PluginExecutionResult:
+    """Reset the active context to a fresh state before any changes have been made."""
+
+    def remove_custom_definitions() -> str:
+        active_context = get_active_context()
+        custom_definitions = [definition for definition in active_context.definitions if definition.source.is_user_editable]
+        active_context.remove_definitions_from_context(custom_definitions)
+        return "Successfully reset the active context"
+
+    with plugin_result(plugin_name, remove_custom_definitions) as result:
+        return result
+
+
+def list_definitions() -> PluginExecutionResult:
+    """
+    List all definitions within the active context.
+
+    Returns:
+        definitions The list of all definitions that are in the active context.
+    """
+    # TODO add implementation here
+    def _implement_and_rename_me():
+        raise NotImplementedError("list_definitions is not implemented.")
+
+    with plugin_result(plugin_name, _implement_and_rename_me) as result:
+        return result
+
+
+def describe_definition(definition_name: str) -> PluginExecutionResult:
+    """
+    Describe a definition in the active context.
+
+    Args:
+        definition_name (str): The name of a defintion in the active context.
+
+    Returns:
+        definition_structure The YAML representation of the structure of the definition.
+        definition_source The source file of the definition in the active context.
+        definition_start The line on which the definition starts in {{active-context.describe-definition.output.definition-source}}.
+    """
+
+    def get_definition_info() -> str:
+        definition = get_active_context().get_definition_by_name(definition_name)
+        if definition is None:
+            raise PluginError(f"{definition_name} is not in the active context.")
+
+        return f"{definition.source.uri}:{definition.lexemes[0].location.line + 1}\n\n{definition.to_yaml()}"
+
+    with plugin_result(plugin_name, get_definition_info) as result:
+        return result
+
+
+def import_state(state_file: str) -> PluginExecutionResult:
+    """
+    Configure the active context to be initialized with the definitions, plugins, etc from a state file.
+
+    Args:
+        state_file (str): The persistent state file from which to get information about how to configure the active context.
+    """
+
+    def import_state_file_if_present() -> str:
+        if not lexists(state_file):
+            raise PluginError(f"The state file {state_file} could not be imported because it doesn't exist.")
+
+        active_context = get_active_context()
+        active_context.clear()
```

Comment:
Currently, the quickest way I found to avoid a duplicate definition error when importing was to do this. I'd still need to debug the issue to see why this is cropping up and how to resolve it.

Reply:
We should be clearing the state when importing files anyway, I've moved this clear call to the LC's `import_from_file` method.